resources/clj-kondo.exports/com.gfredericks/test.chuck/clj_kondo/com/gfredericks/test/chuck/checking.clj

```diff
@@ -13,6 +13,7 @@
     {:node (api/list-node
             (list*
              (api/token-node 'let)
-             (api/vector-node (into [] (:children binding-vec)))
+             (api/vector-node (into [(api/token-node (symbol (gensym "_checking-desc"))) desc]
```

Comment:
what's special about `desc` that requires a call to gensym when nothing else here does?

(I don't know how this stuff works so I'm just curious)

Reply:
`:analyze-call` clj-kondo hooks basically describe how to transform the AST before running the check.
This PR makes `(checking` forms to be transformed (for purposes of `clj-kondo`) as if one wrote
```clojure
(let [_checking-desc32190 <desc>]
  (... <original (checking body))
```
So `clj-kondo` can see that `desc` value is "used", and so are all the values that make it (due to how transitive detection of used stuff works in `clj-kondo`), and we don't get an unused import alert for anything used in `desc`.

I'm `gensym`-ing the name for the binding here just to try to minimise potential clashes with the body of `(checking`. My thinking was that binding `desc` to some constant name could cause issues if enclosing scope binds the same name, for example, if we were using `foo`:
```clojure
(let [foo <some-value>]
  (checking ... <body that uses foo binding>))
```
=> would get analysed by clj-kondo as
```clojure
(let [foo <some-value>]
  (let [foo <desc>]
    <body that uses foo binding... but whoops, our macro expansion broke it!>))
```

Reply:
makes sense, thanks